### PR TITLE
SingleTraitInsertPerStatement - fix formatting for multiline "use"

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4709

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
What does it do with the following example:

```php
<?php
final class Example
{
    use Foo,
        Bar,
//         Bazz,
        Baz;
}
```

**Comment by kubawerlos** [comment]
@GrahamCampbell now it does well

**Comment by GrahamCampbell** [comment]
There seems to still be an issue:

image

The newline in between the `use` and `public function` is getting deleted.

**Comment by kubawerlos** [comment]
@GrahamCampbell where is this happening? Because definitely not in `SingleTraitInsertPerStatementFixer` in this PR.

**Comment by GrahamCampbell** [comment]
Could you add the following test case:

```php
<?php

class Example
{
    use Foo, Bar;

    public function baz() {}
}
```

it should be fixed to:

```php
<?php

class Example
{
    use Foo;
    use Bar;

    public function baz() {}
}
```

I think your code will actually fix it to:

```php
<?php

class Example
{
    use Foo;
    use Bar;
    public function baz() {}
}

File: tests/Fixer/ClassNotation/SingleTraitInsertPerStatementFixerTest.php [hunk 1]
```diff
@@ -81,6 +81,24 @@ final class Example
 {
     use Foo, Bar ;
 }
+',
+            ],
+            'simple III' => [
+                '<?php
+class Example
+{
+    use Foo;use Bar;
```
Comment (GrahamCampbell) [inline tests/Fixer/ClassNotation/SingleTraitInsertPerStatementFixerTest.php]: Should be 
```
use Foo;
use Bar;
```

not:

```
use Foo;use Bar;
```

**Comment by kubawerlos** [comment]
Added, through the expected result is different from each of your examples - we are not adding any new line if none between imports (as in very first test case), so this what would be: 
```php
<?php
class Example
{
    use Foo;use Bar;

    public function baz() {}
}
```

**Comment by GrahamCampbell** [comment]
Right, so your fixer is fixing to the wrong thing, and then another fixer is breaking up the statements?

File: tests/Fixer/ClassNotation/SingleTraitInsertPerStatementFixerTest.php (hunk 1 above)
Comment (kubawerlos) [reply tests/Fixer/ClassNotation/SingleTraitInsertPerStatementFixerTest.php]: No, it shall not -> see the test case `simple I`.

**Comment by GrahamCampbell** [comment]
// cc @SpacePossum. This fixer is meant to fix to the Symfony and PSR-12 style, but seems to not do that?

**Comment by kubawerlos** [comment]
> Right, so your fixer is fixing to the wrong thing

WHAAAT? Are you kidding? See the issue it's fixing - there is a test case that covers this issue...

See the original tests - they are not meant to break the lines, only to do what the fixer describes to do, which is 'Each trait `use` must be done as single statement.' - STATEMENT, not a line.

> This fixer is meant to fix to the Symfony and PSR-12 style

Since when? It seems like overthinking. We have a fixer to split the imports into separate lines since a long time ago, this fixer is meant to be run before it and @SpacePossum thought about it as he wrote this integration test: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.16.1/tests/Fixtures/Integration/priority/single_trait_insert_per_statement,braces.test

**Comment by GrahamCampbell** [comment]
> See the original tests - they are not meant to break the lines,

Yes, I know. I am saying I think the common sense thing to do is to have the fixer split onto new lines, even though the original implementation didn't do this. This is why I have cced in the original fixer author.

**Comment by GrahamCampbell** [comment]
Because of exactly the problem I am describing, with the example I provided. Trying to have separation of responsibilities just leads to it being harder to use, and producing output that is not desirable.

**Comment by GrahamCampbell** [comment]
Though, maybe this is something we should discuss for **after** this PR. ;)

**Comment by GrahamCampbell** [comment]
This PR itself looks fine.

**Comment by SpacePossum** [comment]
> This fixer is meant to fix to the Symfony and PSR-12 style,

No, it is meant/designed to make multiple use statements

> Trying to have separation of responsibilities just leads to it being harder to use, and producing output that is not desirable.

I disagree, please provide more than an opinion and I might change my mind. 

For now, lets leave this out of the PR, the only issue I've seen so far is that there is no priority test between this fixer and `class_attributes_separation`.

**Comment by SpacePossum** [comment]
Thank you @kubawerlos.

**Pull request merged**